maintenance/measure_tile_performance/measure_tile_performance.py

```diff
@@ -7,97 +7,135 @@
 import pytz
 from pytz import timezone
 import os
+import psycopg2
+import statistics
+
+NAME = 'snowplow'
+HOST = 'redshift.analytics.gov.bc.ca'
+PORT = 5439
+
+# connects with redshift database using environment variables
+# set redshift cache to 'off' before running any looker API calls
+# run the API calls
+# set redshift cache back to 'on' after the queries are processed.
+# close the redshift connection
+def main():
+    
+    user = os.environ['lookeruser_rs']
+    passwd = os.environ['lookerpass_rs']
+    connection_string = (
+        f"dbname='{NAME}' "
+        f"host='{HOST}' "
+        f"port='{PORT}' "
+        f"user='{user}' "
+        f"password={passwd}")
+    
+    conn = psycopg2.connect(dsn=connection_string)
 
+    with conn:
+        with conn.cursor() as curs:
+            curs.execute("ALTER USER looker SET enable_result_cache_for_session TO off;")
 
```

Comment:
You are performing this very early in the job. If the argparsing fails, or if in any step prior to the loop there is a failure that causes the program to halt, there is risk that the `enable_result_cache_for_session` will remain `off` for Looker (since the later "`on`" may not be reached).